api-types/index.ts

```diff
@@ -38,30 +38,37 @@ export type DesktopAPI = {
     onBrowserHistoryPush: (listener: (pathName: string) => void) => () => void;
     sendBrowserHistoryPush: (path: string) => void;
 
-    // Calls widget
-    openLinkFromCallsWidget: (url: string) => void;
-    openScreenShareModal: () => void;
-    onScreenShared: (listener: (sourceID: string, withAudio: boolean) => void) => () => void;
-    callsWidgetConnected: (callID: string, sessionID: string) => void;
-    onJoinCallRequest: (listener: (callID: string) => void) => () => void;
-    resizeCallsWidget: (width: number, height: number) => void;
-    focusPopout: () => void;
-    leaveCall: () => void;
-    sendCallsError: (err: string, callID?: string, errMsg?: string) => void;
-
-    // Calls plugin
-    getDesktopSources: (opts: DesktopSourcesOptions) => Promise<DesktopCaptureSource[]>;
-    onOpenScreenShareModal: (listener: () => void) => () => void;
-    shareScreen: (sourceID: string, withAudi: boolean) => void;
+    // Calls
     joinCall: (opts: {
         callID: string;
         title: string;
         rootID: string;
         channelURL: string;
     }) => Promise<{callID: string; sessionID: string}>;
-    sendJoinCallRequest: (callId: string) => void;
+    leaveCall: () => void;
+
+    callsWidgetConnected: (callID: string, sessionID: string) => void;
+    resizeCallsWidget: (width: number, height: number) => void;
+
+    sendCallsError: (err: string, callID?: string, errMsg?: string) => void;
     onCallsError: (listener: (err: string, callID?: string, errMsg?: string) => void) => () => void;
 
+    getDesktopSources: (opts: DesktopSourcesOptions) => Promise<DesktopCaptureSource[]>;
+
+    openScreenShareModal: () => void;
+    onOpenScreenShareModal: (listener: () => void) => () => void;
+
+    shareScreen: (sourceID: string, withAudio: boolean) => void;
+    onScreenShared: (listener: (sourceID: string, withAudio: boolean) => void) => () => void;
+
+    sendJoinCallRequest: (callId: string) => void;
+    onJoinCallRequest: (listener: (callID: string) => void) => () => void;
+
+    openLinkFromCalls: (url: string) => void;
+    onOpenLinkFromCalls: (listener: (url: string) => void) => () => void;
```

Comment:
So is this the calls widget (running standalone) sending an event to the calls plugin to open a link?

Reply:
Yes, widget/popout window depending on the use case.

Reply:
Is there a reason we can't just send this through the Desktop App's method to open a deep link? If we have the server configured, it should just forward the link like normal.

Unless the plugin needs to be aware when the calls widget is navigating for some reason, I think that would be preferred.

Reply:
Not sure if there's a reason other than to keep plugin functionality in plugin as much as possible. I'd have to check how it would look like to go through Desktop. So far we need this event for internal navigation, switching to and focusing on channels where calls are happening essentially.

Reply:
If all we need to do is just tell the web app to navigate somewhere when the calls widget wants it to, I think we can avoid the listener since we can just push up an event to the Desktop App with a path to tell the server to navigate.

You can look at `ViewManager.handleDeepLink` and you can have the `callsWidgetWindow` class just call that instead of forwarding the click to the plugin.

Reply:
@devinbinnie To confirm, this behaviour is pretty much what `handleCallsWidgetChannelLinkClick` was already doing, which you deprecated. Not sure if there's more that we need to add but forwarding to the app wouldn't be necessary any longer for this event.

Reply:
I think that one was forwarding to the app via the `browser_history` module, which we deprecated. This would be using the same method that the deep linking uses, which is basically the same event anyways.

I might be mistaken though.

Reply:
I don't think it was forwarded but one important difference is that `handleCallsWidgetChannelLinkClick` is purely focusing on the case of switching to the channel ID for the current call. Now we need this to be dynamic based on the input so maybe some extra checks are needed to make sure we don't try to navigate to random stuff. I'll look into `handleDeepLink`, thanks!

Reply:
@devinbinnie Updated. So far I think we'd be sending relative URLs only for channel switching but I think that handling full URLs through `ViewManager.handleDeepsLink` could be useful in the future so I added it. Let me know if that makes sense. I updated the other side as well in https://github.com/mattermost/mattermost-plugin-calls/pull/602/commits/6852a5f1453a959cda57c0176893a27e964bb3b8